src/OpenTelemetry.Exporter.Zipkin/ZipkinTraceExporter.cs

```diff
@@ -131,20 +145,51 @@ internal ZipkinSpan GenerateSpan(SpanData otelSpan, ZipkinEndpoint defaultLocalE
                 spanBuilder.ParentId(this.EncodeSpanId(otelSpan.ParentSpanId));
             }
 
+            string peerService = null; // RemoteEndpoint.ServiceName primary.
```

Comment:
This could be simplified to just be a single string and only overwrite if null - essentially moving the logic on line 229 to maintain a single value.

Reply:
@MikeGoldsmith That logic would tweak the behavior slightly in that, depending on the order of the attributes and the content, the last match would win. I updated with a more efficient solution that does what you were suggesting in spirit but also preserves the "priority" that was in there. See below.